packages/ifpack2/src/Ifpack2_SparseContainer_def.hpp

```diff
@@ -124,6 +124,70 @@ void SparseContainer<MatrixType,InverseType>::setParameters(const Teuchos::Param
   List_ = List;
 }
 
+//==============================================================================
+template<class MatrixType, class InverseType>
+void SparseContainer<MatrixType,InverseType>::findRowIndicesAndCounts(
+  int blockIndex,                        // block being processed
+  local_ordinal_type rowIndex,           // row being processed in this block
+  const Teuchos::ArrayView<const local_ordinal_type> &localRows,  
+  const Teuchos::ArrayView<local_ordinal_type> &Indices,
+                                                // Storage for row indices 
+                                                // of inputMatrix
+  const Teuchos::ArrayView<scalar_type> &Values,// Storage for row values 
+                                                // of inputMatrix
+  Teuchos::Array<InverseGlobalOrdinal> &Indices_insert,  // Output: Indices to 
+                                                         // be inserted for
+                                                         // this row
+  Teuchos::Array<InverseScalar> &Values_insert, // Output: Value associated 
+                                                // with the inserted indices
+  size_t &num_entries_found                     // Output: number of Indices 
+                                                // to be inserted
+) const
+{
+  const InverseLocalOrdinal INVALID =
+        Teuchos::OrdinalTraits<InverseLocalOrdinal>::invalid ();
+  const size_t MatrixInNumRows = this->inputMatrix_->getNodeNumRows ();
+  const local_ordinal_type localRow = 
+        this->partitions_[this->partitionIndices_[blockIndex] + rowIndex];
+  size_t numEntries;
+  this->inputMatrix_->getLocalRowCopy(localRow,Indices(),Values(),numEntries);
+
+  num_entries_found = 0;
+  for(size_t k = 0; k < numEntries; ++k)
+  {
+    const local_ordinal_type localCol = Indices[k];
+
+    // Skip off-process elements
+    //
+    // FIXME (mfh 24 Aug 2013) This assumes the following:
+    //
+    // 1. The column and row Maps begin with the same set of
+    //    on-process entries, in the same order.  That is,
+    //    on-process row and column indices are the same.
+    // 2. All off-process indices in the column Map of the input
+    //    matrix occur after that initial set.
+    if(static_cast<size_t> (localCol) >= MatrixInNumRows)
+      continue;
+
+    // for local column IDs, look for each ID in the list
+    // of columns hosted by this object
+    InverseLocalOrdinal jj = INVALID;
+    for(local_ordinal_type kk = 0; kk < this->blockRows_[blockIndex]; kk++)
```

Comment:
In the future, we could consider using `std::find` for this, but not for now : - ) .